File: app/src/main/java/com/example/javarushproject/model/Cat.java
package com.example.javarushproject.model;

import androidx.annotation.NonNull;

import java.util.Objects;

public class Cat extends Animal {
    private int lives;

    public Cat(int height, int weight, String name, int lives) {
        super(height, weight, name);
        this.lives = lives;
    }

    public int getLives() {
        return lives;
    }

    @Override
    public String toString() {
        return "Cat{" +
                "height=" + super.getHeight() +
                ", weight=" + super.getWeight() +
                ", name=" + super.getName() +
                ", lives=" + lives +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Cat cat = (Cat) o;
        return lives == cat.lives;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), lives);
    }
}

File: app/src/main/java/com/example/javarushproject/model/Cat.kt
package com.example.javarushproject.model

import java.util.*

class Cat(height: Int, weight: Int, name: String?, private val lives: Int) : Animal(height, weight, name!!) {

    override fun toString(): String {
        return "Cat{" +
                "height=" + super.height +
                ", weight=" + super.weight +
                ", name=" + super.name +
                ", lives=" + lives +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val cat = other as Cat
        return lives == cat.lives
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), lives)
    }
}